test/e2e/azure_test.go

```diff
@@ -55,8 +58,21 @@ var _ = Describe("CAPZ e2e tests", func() {
 		})
 
 		AfterEach(func() {
+			groupName := input.Management.GetName()
+
 			By("cleaning up e2e resources")
 			ensureCAPZArtifactsDeleted(input)
+
+			By("Making sure there is no leftovers")
```

Comment:
```suggestion
			By("Making sure there are all Azure resources are deleted")
```

Reply:
let's move this line and the below lines to inside `ensureCAPZArtifactsDeleted` and maybe rename the function to something a little more descriptive like `CleanupE2EResources`.

Other than that lgtm

Reply:
sgtm